cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -1127,9 +1141,16 @@ impl Inst {
                     sink.add_call_site(info.opcode);
                 }
                 sink.add_reloc(Reloc::RiscvCallPlt, &info.dest, 0);
-                if let Some(s) = state.take_stack_map() {
+
+                let (stack_map, user_stack_map) = state.take_stack_map();
+                if let Some(s) = stack_map {
                     sink.add_stack_map(StackMapExtent::UpcomingBytes(8), s);
                 }
+                if let Some(s) = user_stack_map {
+                    let offset = sink.cur_offset() + 8;
```

Comment:
(likewise here as above; coupling feels even a little more brittle since we're depending on the fact we emit two four-byte instructions below)